src/lib/react/useRoomConnection.ts

```diff
@@ -354,94 +369,148 @@ export default function useRoomConnection(
-        });
-
-        roomConnection.addEventListener("participant_left", (e) => {
-            const { participantId } = e.detail;
-            dispatch({ type: "PARTICIPANT_LEFT", payload: { participantId } });
-        });
-
-        roomConnection.addEventListener("participant_stream_added", (e) => {
-            const { participantId, stream } = e.detail;
-            dispatch({ type: "PARTICIPANT_STREAM_ADDED", payload: { participantId, stream } });
-        });
-
-        roomConnection.addEventListener("room_connection_status_changed", (e) => {
-            const { roomConnectionStatus } = e.detail;
-            dispatch({ type: "ROOM_CONNECTION_STATUS_CHANGED", payload: { roomConnectionStatus } });
-        });
-
-        roomConnection.addEventListener("room_joined", (e) => {
-            const { localParticipant, remoteParticipants, waitingParticipants } = e.detail;
-            dispatch({ type: "ROOM_JOINED", payload: { localParticipant, remoteParticipants, waitingParticipants } });
-        });
-
-        roomConnection.addEventListener("participant_video_enabled", (e) => {
-            const { participantId, isVideoEnabled } = e.detail;
-            dispatch({ type: "PARTICIPANT_VIDEO_ENABLED", payload: { participantId, isVideoEnabled } });
-        });
-
-        roomConnection.addEventListener("participant_metadata_changed", (e) => {
-            const { participantId, displayName } = e.detail;
-            dispatch({ type: "PARTICIPANT_METADATA_CHANGED", payload: { participantId, displayName } });
-        });
-
-        roomConnection.addEventListener("screenshare_started", (e) => {
-            const { participantId, id, hasAudioTrack, stream } = e.detail;
-
-            dispatch({ type: "SCREENSHARE_STARTED", payload: { participantId, id, hasAudioTrack, stream } });
-        });
-
-        roomConnection.addEventListener("screenshare_stopped", (e) => {
-            const { participantId, id } = e.detail;
-            dispatch({ type: "SCREENSHARE_STOPPED", payload: { participantId, id } });
-        });
-
-        roomConnection.addEventListener("streaming_started", (e) => {
-            const { status, startedAt } = e.detail;
-            dispatch({ type: "STREAMING_STARTED", payload: { status, startedAt } });
-        });
-
-        roomConnection.addEventListener("streaming_stopped", () => {
-            dispatch({ type: "STREAMING_STOPPED" });
-        });
-
-        roomConnection.addEventListener("waiting_participant_joined", (e) => {
-            const { participantId, displayName } = e.detail;
-            dispatch({ type: "WAITING_PARTICIPANT_JOINED", payload: { participantId, displayName } });
-        });
+    const eventListeners = React.useMemo(
+        () =>
+            [
+                {
+                    eventName: "chat_message",
+                    listener: (e) => {
+                        const chatMessage = e.detail as ChatMessage;
```

Comment:
```suggestion
                    listener: (e: CustomtEvent<ChatMessage>) => {
                        const chatMessage = e.detail;
```

Can't we do a type declaration here instead of asserting with `as`?

Reply:
This is a good idea, thanks 🙏